daml.mk

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2021, Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+comma := ,
+daml ?= daml
+rdars := $(shell scripts/getDamlProjects.sh)
+dars := $(patsubst %$(comma)js,%,$(patsubst %$(comma),%,$(rdars)))
+jsdars := $(patsubst %$(comma)js,%,$(filter %$(comma)js,$(rdars)))
+project-root = $(patsubst %/.daml/dist,%,$(@D))
+phony-test-targets := $(patsubst %,%-phony-test-target,$(dars))
+
+ifneq ($(.SHELLSTATUS),0)
+  $(error shell command failed)
+endif
+
+.PHONY: build-dars
+build-dars: $(dars)
+
+.PHONY: test-dars
+test-dars: $(phony-test-targets)
+
+ui/daml.js: $(jsdars)
+	$(daml) codegen js $^ -o $@
+	@touch $@
+
+.PHONY: $(phony-test-targets)
+$(phony-test-targets): build-dars
+	$(daml) test --project-root $(project-root) --junit $(PWD)/target/$(@F).xml
+
+.SECONDEXPANSION:
```

Comment:
(this is the magic, idea borrowed from the original example. pls see make manual for this if you are interested.)

Reply:
we can replace it later, automatically creating the dependency file by make, as in the current version